### DISCUSSION TITLE
WIT data is available

### DISCUSSION THREAD

**Discussion by sorrge** [lucidrains/DALLE-pytorch#104] [General]
11M images and captions: https://github.com/google-research-datasets/wit/blob/main/DATA.md

**Comment by afiaka87** [comment]
## Download Links

Here are the links to download the 10 files. Each one is a TSV with the links required for download along with captions and metadata. 
[wit_v1.train.all-00000-of-00010.tsv.gz](https://storage.googleapis.com/gresearch/wit/wit_v1.train.all-00000-of-00010.tsv.gz)

[wit_v1.train.all-00001-of-00010.tsv.gz](https://storage.googleapis.com/gresearch/wit/wit_v1.train.all-00001-of-00010.tsv.gz)

[wit_v1.train.all-00002-of-00010.tsv.gz](https://storage.googleapis.com/gresearch/wit/wit_v1.train.all-00002-of-00010.tsv.gz)

[wit_v1.train.all-00003-of-00010.tsv.gz](https://storage.googleapis.com/gresearch/wit/wit_v1.train.all-00003-of-00010.tsv.gz)

[wit_v1.train.all-00004-of-00010.tsv.gz](https://storage.googleapis.com/gresearch/wit/wit_v1.train.all-00004-of-00010.tsv.gz)

[wit_v1.train.all-00005-of-00010.tsv.gz](https://storage.googleapis.com/gresearch/wit/wit_v1.train.all-00005-of-00010.tsv.gz)

[wit_v1.train.all-00006-of-00010.tsv.gz](https://storage.googleapis.com/gresearch/wit/wit_v1.train.all-00006-of-00010.tsv.gz)

[wit_v1.train.all-00007-of-00010.tsv.gz](https://storage.googleapis.com/gresearch/wit/wit_v1.train.all-00007-of-00010.tsv.gz)

[wit_v1.train.all-00008-of-00010.tsv.gz](https://storage.googleapis.com/gresearch/wit/wit_v1.train.all-00008-of-00010.tsv.gz)

[wit_v1.train.all-00009-of-00010.tsv.gz](https://storage.googleapis.com/gresearch/wit/wit_v1.train.all-00009-of-00010.tsv.gz)

**Reply by afiaka87** [reply]
@lucidrains I'm gonna (try to) download them all, resize to 256x256 with jpeg compression at 95% and reupload them somewhere suitable for us.

**Reply by TheodoreGalanos** [reply]
nice thank you! I wonder if an interface to sample specific topics (out of the text descriptions) would be nice. 4m images is cool to train a model on, but I wonder if we could slice those to create specific datasets.

**Reply by afiaka87** [reply]
We definitely will need the slicing even _if_ we use the whole dataset as we're going to be trying to generate a variety of captions based on which topic it is in.

**Comment by sorrge** [comment]
I counted ~5.4M unique images with English captions (those on "en" pages).

**Reply by Ki6an** [reply]
![image](https://user-images.githubusercontent.com/63173962/111849117-d92e6a80-8932-11eb-82b4-9df9666b54ac.png)
![image](https://user-images.githubusercontent.com/63173962/111849225-12ff7100-8933-11eb-9cbd-cc0c346f2104.png)

here's some more info from the [paper](https://arxiv.org/pdf/2103.01913.pdf).

**Reply by sorrge** [reply]
Yeah, I must have messed up the count and didn't exclude the repeated URLs. So, down to ~4M. Still a lot.

**Reply by Ki6an** [reply]
are all English data spread out throughout the whole 10 files? 
translating other (popular) language text to English would give more data. Wit still has lots of data in german(2.6M), French(2M), Russian(1.5M), etc.
not sure if they're the same caption for an image in different languages.

**Reply by sorrge** [reply]
Yes, they are all mixed up randomly. And yes, there are many images without any English caption, which could be used with a translated caption.

**Reply by Ki6an** [reply]
just curious, how long would it roughly take to train `dall-e` on this (~4m) dataset?

**Reply by sorrge** [reply]
@afiaka87 could tell, because he already trained a large~ish model.

**Reply by TheodoreGalanos** [reply]
Well I'm not sure it's enough for 1 epoch training, maybe a few weeks on one A100?

**Reply by afiaka87** [reply]
> just curious, how long would it roughly take to train `dall-e` on this (~4m) dataset?
@Ki6an 

It really depends entirely on your batch_size, depth, . If you're on an A100 and can use a batch_size of 32 (depth 64, heads 16, full attention) you can actually finish a full epoch on ~200,000 images (COCO in my case) in far less than a day. 

I implore you to go check out my various training attempts though. They dont give much of a result, but they will _definitely_ inform you of the runtimes necessary at various DEPTH, HEADS, BATCH_SIZE, etc.
